File: app/Syncthing/Noauth.py
from .BaseAPI import BaseAPI

class Noauth(BaseAPI):
    """ HTTP REST endpoint for Noauth Services.

        Implements endpoints of https://docs.syncthing.net/dev/rest.html#noauth-endpoints
    """

    prefix = '/rest/noauth/'

    def health(self) -> bool:
        """ Returns true if the server replies with a {"status": "OK"} object.

            Returns:
                bool
        """
        return self.get('health').get('status', '') == 'OK'

File: app/src/syncthing/Noauth.ts
import RequestBase from './RequestBase';

export default class Noauth extends RequestBase {
  /**
   * HTTP REST endpoint for Noauth Services.
   * Implements endpoints of https://docs.syncthing.net/dev/rest.html#noauth-endpoints
   */
  protected prefix = '/rest/noauth/';

  /**
   * Returns true if the server replies with a {"status": "OK"} object.
   */
  async health(): Promise<boolean> {
    const data = await this.get<{ status: string }>('health');
    return data.status === 'OK';
  }
}
